plugins/agents/ping/Ping.go

```diff
@@ -0,0 +1,113 @@
+package ping
+
+import (
+	"strings"
+	"time"
+
+	"github.com/abrander/agento/plugins"
+	"github.com/abrander/agento/timeseries"
+	gping "github.com/gansoi/gansoi/plugins/agents/ping"
+)
+
+type Data struct {
+	Loss    int    `json:"loss"`
+	Sent    int    `json:"sent"`
+	Replies int    `json:"replies"`
+	TimeAvg int64  `json:"timeavg"`
+	TimeMin int64  `json:"timemin"`
+	TimeMax int64  `json:"timemax"`
+	Ip      string `json:"server"`
+}
+
+type Ping struct {
+	Data   []Data `json:"data"`
+	pinger *gping.ICMPService
+
+	Ip    string `toml:"ip" json:"ip" description:"The ip(s) to ping (multiple can be separated by comma)"`
+	Count int    `toml:"count" json:"count" description:"Number of packages to send"`
+}
+
+func init() {
+	plugins.Register("ping", NewPing)
+}
+
+func NewPing() interface{} {
+	return new(Ping)
+}
+
+func (p *Ping) Gather(transport plugins.Transport) error {
+	// Create and start the ICMP pinger service
+	if p.pinger == nil {
+		p.pinger = gping.NewICMPService()
+		p.pinger.Start()
+	}
+
+	// Default to "one ping only" if nothing else is specified in config
+	count := 1
+	if p.Count > 0 {
+		count = p.Count
+	}
+
+	ips := strings.Split(p.Ip, ",")
+	for _, ip := range ips {
+		data := Data{}
```

Comment:
Can we insert a `ip = strings.TrimSpace(ip)` here?

Reply:
@abrander done